webapp/src/components/ui/MatrixView/MatrixGraphView.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from 'react';
+import Plot from 'react-plotly.js';
+import { createStyles, makeStyles, Theme, InputLabel, FormControl, Box, OutlinedInput, Chip, Select, MenuItem, FormControlLabel, Checkbox } from '@material-ui/core';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import { useTranslation } from 'react-i18next';
+import { MatrixType } from '../../../common/types';
+import 'handsontable/dist/handsontable.min.css';
+
+const useStyles = makeStyles((theme: Theme) => createStyles({
+  root: {
+    width: '100%',
+    height: '100%',
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+
+  form: {
+    minWidth: '200px',
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+  },
+  input: {
+    top: '-7px',
+    left: '14px',
+  },
+  container: {
+    width: '100%',
+    display: 'flex',
+    alignItems: 'center',
+  },
+  box: {
+    marginRight: theme.spacing(1),
+    height: '24px',
+    '& span': {
+      padding: '8px',
+    },
+  },
+  select: {
+    '& .MuiSelect-select': {
+      paddingBottom: theme.spacing(1) + 4,
+      minWidth: '200px',
+    },
+  },
+  monotonous: {
+    marginLeft: theme.spacing(2),
+  },
+  autosizer: {
+    display: 'block',
+    width: '100%',
+    height: '100%',
+  },
+}));
+
+interface PropTypes {
+  matrix: MatrixType;
+}
+
+export default function MatrixGraphView(props: PropTypes) {
+  // eslint-disable-next-line react/destructuring-assignment
+  const [t] = useTranslation();
+  const { matrix } = props;
+  const { data = [], columns = [], index = [] } = matrix;
+  const classes = useStyles();
+  const [columnName, setColumnName] = useState<string[]>([]);
+  const [monotonous, setMonotonous] = useState<boolean>(false);
+
+  const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setColumnName(event.target.value as string[]);
+  };
+
+  const monotonousChange = () => {
+    setMonotonous(!monotonous);
+  };
+
+  const unitChange = (tabBase: Array<number>) => {
+    const tab = [...tabBase];
+    for (let i = 0; i < tab.length; i += 1) {
+      if (i === 0) {
+        tab.splice(i, 1, 100 / tab.length);
+      } else {
+        tab.splice(i, 1, 100 / tab.length + tab[i - 1]);
+      }
+    }
```

Comment:
Here, we compute "100/tab.length" tab.length times ! we should compute it once.
Also, this is an arithmetic progression 
```
U0 = a
Un = Un-1 + a
```
So we have the general term `Un = a(n+1)`

This could be then simplified with:
```
const stepLength = 100/tab.length;
return tabBase.map((el, index) => stepLength * (index + 1));
```